File: ios/BatteryHolder/Models/BatteryReading.swift
import Foundation

/// A raw sample as delivered by a transport (BLE notify or Wi-Fi poll),
/// before the app applies the pin configuration.
struct DeviceSample: Equatable {
    let rawADC: Int
    /// Voltage the device itself computed, if it reports one.
    let deviceVolts: Double?
    let pinId: String?
    let timestamp: Date

    init(rawADC: Int, deviceVolts: Double? = nil, pinId: String? = nil, timestamp: Date = Date()) {
        self.rawADC = rawADC
        self.deviceVolts = deviceVolts
        self.pinId = pinId
        self.timestamp = timestamp
    }
}

/// A fully-interpreted reading the UI displays and charts.
struct BatteryReading: Identifiable, Equatable {
    let id = UUID()
    let timestamp: Date
    let rawADC: Int
    let voltage: Double
    /// State of charge 0...1.
    let percentage: Double?
    let pinId: String
}

File: android_app/lib/models/battery_reading.dart
/// A raw sample as delivered by a transport (BLE notify or Wi-Fi poll),
/// before the app applies the pin configuration.
class DeviceSample {
  final int rawADC;

  /// Voltage the device itself computed, if it reports one.
  final double? deviceVolts;
  final String? pinId;
  final DateTime timestamp;

  DeviceSample({
    required this.rawADC,
    this.deviceVolts,
    this.pinId,
    DateTime? timestamp,
  }) : timestamp = timestamp ?? DateTime.now();
}

/// A fully-interpreted reading the UI displays and charts.
class BatteryReading {
  final DateTime timestamp;
  final int rawADC;
  final double voltage;

  /// State of charge 0...1.
  final double? percentage;
  final String pinId;

  const BatteryReading({
    required this.timestamp,
    required this.rawADC,
    required this.voltage,
    required this.percentage,
    required this.pinId,
  });
}
